hack/upgrade.sh

```diff
@@ -36,37 +36,30 @@ go generate ./...
 go run cmd/createorupdate/createorupdate.go
```

Comment:
ok, so now we have RP simulation of 

1.  new manifest comes in
2.  rp calls validateInternal (when merged)
3.  rp calls generate
4.  rp calls generateARM??  
5.  rp calls vmssrollout??
6.  new sync pod is deployed

I'm not clear on steps 4 and 5, I think we need to nail down the flow there.  It seems like we're making the assumption that ARM never changes and it's only vmss updates.

Reply:
I don't want to put words in @jim-minter's mouth but from what I inferred from our last conversation was that for PP it would be enough to have scaleset upgrades and we could fix applying the rest of the ARM template before GA. Even with something that applies the ARM template in place, we still need vmssrollout to manage VMSS rolling upgrades, since we cannot use the built-in rolling upgrade feature for a couple of reasons (lack of HTTPS readiness probes, and I am not sure whether in-place upgrades are supported).

Reply:
FWIW, I can't see how acs-engine applies azuredeploy.json changes on upgrades. They seem to be doing mere VM upgrades, too.

Reply:
I think if case like this appears we will just need to deploy new arm template. Should we move from `azuredeploy` as a name inside ARM to something like `OSA-Vxxxx`? This would allow us to have better tracking of arm deployed artifacts?